tests/workflows/pw/test_base.py

```diff
@@ -40,29 +39,59 @@ def test_handle_unrecoverable_failure(generate_workchain_pw):
     assert result == PwBaseWorkChain.exit_codes.ERROR_UNRECOVERABLE_FAILURE
 
 
-def test_handle_out_of_walltime(generate_workchain_pw):
+def test_handle_out_of_walltime(generate_workchain_pw, fixture_localhost, generate_remote_data):
     """Test `PwBaseWorkChain.handle_out_of_walltime`."""
-    process = generate_workchain_pw(exit_code=PwCalculation.exit_codes.ERROR_OUT_OF_WALLTIME)
+    remote_data = generate_remote_data(computer=fixture_localhost, remote_path='/come_to_papa')
+    process = generate_workchain_pw(
+        exit_code=PwCalculation.exit_codes.ERROR_OUT_OF_WALLTIME, pw_outputs={'remote_folder': remote_data}
+    )
     process.setup()
 
+    result = process.handle_electronic_convergence_not_achieved(process.ctx.children[-1])
     result = process.handle_out_of_walltime(process.ctx.children[-1])
     assert isinstance(result, ProcessHandlerReport)
+    assert process.ctx.inputs.parameters['CONTROL']['restart_mode'] == 'restart'
     assert result.do_break
 
     result = process.inspect_process()
     assert result.status == 0
 
 
-def test_handle_electronic_convergence_not_reached(generate_workchain_pw):
+def test_handle_out_of_walltime_structure_changed(generate_workchain_pw, generate_structure):
+    """Test `PwBaseWorkChain.handle_out_of_walltime`."""
+    structure = generate_structure()
+    process = generate_workchain_pw(
+        exit_code=PwCalculation.exit_codes.ERROR_OUT_OF_WALLTIME, pw_outputs={'output_structure': structure}
+    )
+    process.setup()
+
+    result = process.handle_out_of_walltime(process.ctx.children[-1])
+    assert isinstance(result, ProcessHandlerReport)
+    assert process.ctx.inputs.parameters['CONTROL']['restart_mode'] == 'from_scratch'
+    assert result.do_break
+
+    result = process.inspect_process()
+    assert result.status == 0
+
+
+def test_handle_electronic_convergence_not_achieved(generate_workchain_pw, fixture_localhost, generate_remote_data):
     """Test `PwBaseWorkChain.handle_electronic_convergence_not_achieved`."""
-    process = generate_workchain_pw(exit_code=PwCalculation.exit_codes.ERROR_ELECTRONIC_CONVERGENCE_NOT_REACHED)
+    remote_data = generate_remote_data(computer=fixture_localhost, remote_path='/come_to_papa')
```

Comment:
ohhh boy

Reply:
At 3 am I though this was quite funny. 😅

Reply:
I would try to keep the jokes to a PG rating though, haha.